### PR TITLE
NIFI-5188: DruidTranquilityController does not fully support Druid aggregator

### PR DESCRIPTION
**Pull request by dkhwangbo**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
  - I think verify this PR is enough with unit test already used.
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by dkhwangbo** [comment]
@joewitt @pvillard31 @mattyb149 Any ideas?

**Comment by mattyb149** [comment]
The code changes LGTM, but I haven't built/tried them yet. I see quite a few [changes in 0.9.2](https://github.com/druid-io/druid/pulls?utf8=%E2%9C%93&q=is%3Apr%20is%3Aclosed%20milestone%3A0.9.2), do any of them not work with our current version of Tranquility?

**Comment by mattyb149** [comment]
@vakshorton do you mind taking a look at the changes?

**Comment by dkhwangbo** [comment]
@mattyb149 Hi. Thanks for your comment. 
I already use this patch builded with Druid 0.9.2 for a week and no abnormal operation is detected. But I think Using Druid 0.9.1 is appropriate way because latest version of tranquility is builded against Druid 0.9.1(refer: [tranquility 0.8.2 release note](https://github.com/druid-io/tranquility/releases/tag/v0.8.2)).
I rollback this change. Please review.

**Comment by dkhwangbo** [comment]
@joewitt @pvillard31 @mattyb149 Any ideas?

**Comment by mattyb149** [comment]
Verified L&N so far, the only unaccounted for dependency is Yahoo's sketches-core, which is ASL 2.0 licensed with no NOTICE: https://github.com/DataSketches/sketches-core

**Comment by dkhwangbo** [comment]
@mattyb149 Manual modification is only way to update LICENSE files? Any tools or scripts to do with?

File: nifi-nar-bundles/nifi-druid-bundle/nifi-druid-controller-service/src/main/java/org/apache/nifi/controller/druid/DruidTranquilityController.java [hunk 1]
```diff
@@ -518,56 +515,23 @@ public String getTransitUri() {
...
+        mapper.registerModules(Lists.newArrayList(new SketchModule().getJacksonModules()));
+        mapper.registerModules(Lists.newArrayList(new ApproximateHistogramDruidModule().getJacksonModules()));
 
-    private AggregatorFactory getCountAggregator(Map<String, String> map) {
-        return new CountAggregatorFactory(map.get("name"));
+        try {
+            return mapper.readValue(
+                aggregatorJSON,
+                new TypeReference<List<AggregatorFactory>>()
+                {
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-druid-bundle/nifi-druid-controller-service/src/main/java/org/apache/nifi/controller/druid/DruidTranquilityController.java]: CheckStyle error here, the curly brace should be on the line above. I'll fix this on merge

**Comment by mattyb149** [comment]
+1 LGTM, I was able to verify that the existing aggregators are still present, and that the sketch and histogram aggregator modules are added. Thanks for this improvement! Merging to master

**Pull request closed without merging**